### ISSUE TITLE
[Feature Request] Support for custom parameters (or just User filter on nyaa?)

### ISSUE THREAD

**Issue by 13steinj** [Jackett/Jackett#11567]
Not an error, so I hope feature requests are welcome.

Nyaa has the capability to filter by specific uploaders.

This is useful because albeit some users not being considered "Trusted" on the site, they...well they are trustworthy. I.e., the trusted filter is too restrictive, but leaving things out to the open is possibly dangerous.

So suggestion: support the user filter (and/or custom url paramters). For the user, url parameter u=User, ex, thesite /?page=rss&u=13steinj among any other parameters necessary.

Slight issue with this approach is there's no way to see combine multiple users in one URI, which means either special-casing nyaa or making it such that you can have multiple of certain indexers.

I would assist in contributing code but I both am unfamiliar in C# (not really an issue) and unfamiliar in how I would set up this repo on Linux using Rider.

**Comment by ilike2burnthing** [comment]
This change would need to be specific to the indexer, in this case nyaasi, a YAML indexer, so no C# knowledge needed - https://github.com/Jackett/Jackett/blob/master/src/Jackett.Common/Definitions/nyaasi.yml

The only way I could see this really working would be a text box in the indexer's configuration panel which users could enter a single uploader's name and so search by it. However, this would have to be changed every time they wanted to either search by a different uploader or none at all.

As you mentioned, the alternative to this is to have multiple of the same indexer, each configured slightly differently, but this would have to be done by you (basically creating nyaasi2.yml nyaasi3.yml nyaasi4.yml etc. with the id and name also changed), but any changes made to the base indexer by us would have to be done manually by you for those clones.

**Comment by ilike2burnthing** [comment]
@13steinj any thoughts?

**Comment by 13steinj** [comment]
@ilike2burnthing My apologies I thought you were thinking out loud in terms of how this would be implemented, not asking for thoughts / providing alternatives.

I'm slightly confused-- is the multiple copies of indexers something that can work out of the box as is? If so this would work, but for obvious reasons is cumbersome (and potentially problematic if there are other indexes that have a similar necessity)

Personally if you want my honest opinion it goes a bit overboard but there's two relatively well thought out options that allow for easy extensibility:

* If you're familiar with systemd, one can make template unit files. So possibly we could make nyaa (or other indexers) template indexer(s) rather than regular ones. Jackett would store information somewhere of what templates exist (most easily done by just writing to a folder and reading from it) Or using systems similar to systemd's override.conf files. This way changes to the base indexer could be made, and they would transfer to the new indexer. This limits problems to only when overrideable fields change (for whatever reason) but that can also be marked with a commented timestamp in the override configs for when they exist. 

* add custom field types to indexers. For example, one could have a `forEach` field type (which only matters internally) and Jackett would read that field, separate it (say by commas), fetch from multiple urls filled out with that field each with one item, and merge the results. Other field types may be relevant too, such as a `rewriting` to a specific rule/regexp.

Caveat here though is I'd have to relearn C# as I go (mostly problem being setting up a dev environment in my setup) if I was to work on this.

**Comment by ilike2burnthing** [comment]
Yes, multiple copies of the same indexer currently works, so long as it's a YAML indexer and the name and ID are changed.

PRs are always welcome, though that sounds like a fair bit of work to work around a small issue. As for the environment - https://github.com/Jackett/Jackett/blob/master/CONTRIBUTING.md#contributing-code

**Comment by ilike2burnthing** [comment]
I can add https://github.com/Jackett/Jackett/issues/11567#issuecomment-822900190 if you wanted to try it out first, or you can manually add it yourself (edit the .yml then restart Jackett and access the indexer's config panel).

**Comment by 13steinj** [comment]
@ilike2burnthing I'll be trying it out first for now.

At minimum, regarding copies of the indexer / templating, it may be a good idea to add some functionality to copy the indexer's yaml to a new indexer eg "nyaa-inserttexthere" and "inserttexthere" would be available as a variable?

**Comment by 13steinj** [comment]
To be clear this isn't the only circumstance where multiple indexers of the same "type" are useful, I can think of several private trackers which would benefit here.

**Comment by ilike2burnthing** [comment]
I think just generic `nyaa-2`, `nyaa-3`, etc. will be fine, users can add tags if they want to distinguish them further.
